osu.Game/Overlays/Rankings/HumanizedNumber.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Game.Graphics.Sprites;
+using osu.Framework.Graphics.Cursor;
+
+namespace osu.Game.Overlays.Rankings
+{
+    public class HumanizedNumber : OsuSpriteText, IHasTooltip
+    {
+        private const int k = 1000;
+
+        public string TooltipText => value < k ? "" : $"{value:N0}";
+
+        private readonly long value;
+
+        public HumanizedNumber(long value)
+        {
+            this.value = value;
+
+            Text = numberToText(value);
+        }
+
+        private string numberToText(long value)
+        {
+            var suffixes = new[]
+            {
+                "",
+                "k",
```

Comment:
where are you getting these from? why is thousands "k" but the reset not shortened? also i'm pretty sure there's already an implementation of this exact thing somewhere else in the code base.

Reply:
it's just copy-paste from web side

Reply:
The rest isn't shortened because it's using full names in web
![Screenshot_2](https://user-images.githubusercontent.com/22874522/64937842-8323c700-d864-11e9-97d9-565bc32ab903.png)

Reply:
I could just use `Humanizer.ToMetric`, but it works only for 32bit int and you can't manually modify the text after the number

Reply:
There should be a comment showing where you got the implementation from.